src/models/base.py

```diff
@@ -213,22 +17,4 @@ def get_forward_output_channels(self) -> Union[int, List[int]]:
         Returns: Outpus channels.
         """
         pass
-    
-    def get_output_channels(self) -> Tuple[Union[int, List[int]], List[int]]:
-        """Create forward an hooks channels numbers.
-        
-        Returns:
-            forward_channels: Forward pass output channels.
-            hooks_channels: Hooks output channels.
-        """
-        forward_channels = self.get_forward_output_channels()
-        hooks_channels = self._get_hooks_output_channels()
-        return forward_channels, hooks_channels
-
-    @property
-    def features_info(self):
-        return self.__features_info
-
-    @property
-    def feature_hooks(self):
-        return self.__feature_hooks
+    
```

Comment:
Add abstract method `no_weight_decay`, look for the signature in `Constructor`. And remove the `hasattr` check there